scripts/assets/test_file_exemptions.textproto

```diff
@@ -556,6 +556,12 @@ exempted_file_path: "domain/src/main/java/org/oppia/android/domain/util/FloatExt
 exempted_file_path: "domain/src/main/java/org/oppia/android/domain/util/FractionExtensions.kt"
 exempted_file_path: "domain/src/main/java/org/oppia/android/domain/util/JsonAssetRetriever.kt"
 exempted_file_path: "domain/src/test/java/org/oppia/android/domain/classify/InteractionObjectTestBuilder.kt"
+exempted_file_path: "instrumentation/src/java/org/oppia/android/instrumentation/EndToEndTestGcsResourceModule.kt"
+exempted_file_path: "instrumentation/src/java/org/oppia/android/instrumentation/EndToEndTestImageParsingModule.kt"
+exempted_file_path: "instrumentation/src/java/org/oppia/android/instrumentation/EndToEndTestNetworkModule.kt"
```

Comment:
These modules can be reasonably tests to make sure that their bindings correctly point to the test variants. I suggest adding basic tests for each.

Reply:
Can you provide some more info regarding this.
I mean what exactly do I have to test for each module.

I'm expecting we need a directory `modules` in the javatests sourcesets

Reply:
Created a quick test suite here
https://gist.github.com/FareesHussain/8eb7eb08d5b1f71f683df99cfd91b50e
But I'm unable to run the tests,
Same goes for the robolectric tests of the app module
with a different error
https://gist.github.com/FareesHussain/b3ed165179cf23aff52ab20ac9ee842e

Reply:
Done.
I've added tests for the Modules (EndToEndTestGcsResourceModule and EndToEndTestImageParsingModule)
Not sure how to test the TestApplicationComponent and OppiaTestApplication

Reply:
Seperated the modules and components into a separate directory application similar to the app module